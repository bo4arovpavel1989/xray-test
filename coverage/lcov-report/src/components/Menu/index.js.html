<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Menu/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/Menu</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { withRouter, Link } from 'react-router-dom'
import './Menu.sass'
&nbsp;
class Menu extends React.Component {
  render () {
    return (
      &lt;div className='container'&gt;
        &lt;div className='menuItem'&gt;
          &lt;Link className='menuButton' to='/test'&gt;Начать тест&lt;/Link&gt;
        &lt;/div&gt;
        &lt;div className='menuItem'&gt;
          &lt;Link className='menuButton' to='/admin'&gt;Настройки&lt;/Link&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    )
  }
}
&nbsp;
export default withRouter(Menu)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Apr 18 2019 09:43:57 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
