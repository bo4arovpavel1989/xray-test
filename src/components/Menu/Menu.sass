.menuItem
  display: inline-block
  padding: 20px 40px 20px 0

.menuButton
  width: 200px
  height: 50px
  text-decoration: none
  padding: 10px 20px 10px 20px
  text-align: center
  border: none
  font-weight: bold
  color: #f5f5f5 !important
  background-color: #89f77b
  background-image: linear-gradient(to top right, #89f77b 0% ,#004500 100%)
  box-shadow: 1px 0px 2px #bababa, inset 0px 0px 1px #ffffff
  margin: 5px
  cursor: pointer
