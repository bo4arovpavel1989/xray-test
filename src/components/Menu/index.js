import React from 'react'
import { withRouter } from 'react-router-dom'

class Menu extends React.Component {
  constructor () {
    super()
  }

  render () {
    return (
      <div>
      </div>
    )
  }
}

export default withRouter(Menu)
