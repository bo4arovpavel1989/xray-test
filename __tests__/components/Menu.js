import React from 'react';
import { BrowserRouter } from 'react-router-dom'
import Menu from '../../src/components/Menu';
import Enzyme, { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Menu component', () => {
  it('should render correct', () => {
      const menu = mount(
        <BrowserRouter>
          <Menu/>
        </BrowserRouter>
      );

      expect(shallowToJson(menu)).toMatchSnapshot()
  })
})
