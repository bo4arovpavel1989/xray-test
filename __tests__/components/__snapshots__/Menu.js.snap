// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component should render correct 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Menu)>
      <Route>
        <Menu
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <div
            className="container"
          >
            <div
              className="menuItem"
            >
              <Link
                className="menuButton"
                replace={false}
                to="/test"
              >
                <a
                  className="menuButton"
                  href="/test"
                  onClick={[Function]}
                >
                  Начать тест
                </a>
              </Link>
            </div>
            <div
              className="menuItem"
            >
              <Link
                className="menuButton"
                replace={false}
                to="/admin"
              >
                <a
                  className="menuButton"
                  href="/admin"
                  onClick={[Function]}
                >
                  Настройки
                </a>
              </Link>
            </div>
          </div>
        </Menu>
      </Route>
    </withRouter(Menu)>
  </Router>
</BrowserRouter>
`;
